### ISSUE TITLE
Second Port forwarding process hangs instead of process exiting when binding to same ports

### ISSUE THREAD

**Issue by jcantrill** [openshift/origin#3975]
While testing out port-forwarding in JBT https://issues.jboss.org/browse/JBIDE-20397, which wraps oc binary, trying to port forward to the same pod on the same local and remote ports causes the process to hang.  The following message is displayed but the process does not exit unless you 'cntrl+c':

```
$ oc port-forward -p hello-openshift 8889:8080
E0731 20:20:01.957450   23274 portforward.go:214] Unable to create listener: Error listen tcp4 127.0.0.1:8889: bind: address already in use
E0731 20:20:01.957650   23274 portforward.go:214] Unable to create listener: Error listen tcp6 [::1]:8889: bind: address already in use
W0731 20:20:01.957692   23274 portforward.go:165] Unable to listen on port {8889 8080}: All listeners failed to create with the following errors: listen tcp4 127.0.0.1:8889: bind: address already in use, listen tcp6 [::1]:8889: bind: address already in use

```

It would be helpful if the process exited with a non-zero return code.

**Comment by jcantrill** [comment]
cc @ncdc

**Comment by ncdc** [comment]
Upstream fix: https://github.com/GoogleCloudPlatform/kubernetes/pull/12162

**Comment by danmcp** [comment]
@pmorie

**Comment by ncdc** [comment]
Upstream PR just merged.

**Comment by ncdc** [comment]
Do you all want an UPSTREAM patch for this prior to the next rebase?

**Comment by jcantrill** [comment]
This is not urgent for us as long as we know the solution is coming relatively soon

**Comment by liggitt** [comment]
too late, already pulled in as part of https://github.com/openshift/origin/pull/4346

**Comment by liggitt** [comment]
Fixed in #4346, closing
